packages/analytics-js/src/components/configManager/util/commonUtil.ts

```diff
@@ -101,28 +126,102 @@ const updateStorageState = (logger?: ILogger): void => {
     storageEncryptionVersion = DEFAULT_STORAGE_ENCRYPTION_VERSION;
   }
 
+  // Allow migration only if the configured encryption version is the default encryption version
+  const configuredMigrationValue = storageOptsFromLoad?.migrate;
+  const finalMigrationVal =
+    (configuredMigrationValue as boolean) &&
+    storageEncryptionVersion === DEFAULT_STORAGE_ENCRYPTION_VERSION;
+
```

Comment:
Any reason for this change? only migration to v3.

Reply:
There is no change in this functionality. The block was refactored, so the code moved around.